ts/src/pro/binance.ts

```diff
@@ -886,64 +890,81 @@ export default class binance extends binanceRest {
         /**
          * @method
          * @name binance#watchTickers
+         * @see https://binance-docs.github.io/apidocs/spot/en/#all-market-tickers-stream
+         * @see https://binance-docs.github.io/apidocs/futures/en/#all-market-tickers-streams
+         * @see https://binance-docs.github.io/apidocs/delivery/en/#all-market-tickers-streams
+         * @see https://binance-docs.github.io/apidocs/voptions/en/#24-hour-ticker
          * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for all markets of a specific list
          * @param {string[]} symbols unified symbol of the market to fetch the ticker for
          * @param {object} [params] extra parameters specific to the exchange API endpoint
          * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/#/?id=ticker-structure}
          */
         await this.loadMarkets ();
         symbols = this.marketSymbols (symbols, undefined, true, true, true);
-        const marketIds = this.marketIds (symbols);
-        let market = undefined;
+        let firstMarket = undefined;
         let type = undefined;
-        if (symbols !== undefined) {
-            market = this.market (symbols[0]);
+        if (!this.isEmpty (symbols)) {
+            firstMarket = this.market (symbols[0]);
         }
-        [ type, params ] = this.handleMarketTypeAndParams ('watchTickers', market, params);
+        [ type, params ] = this.handleMarketTypeAndParams ('watchTickers', firstMarket, params);
         let subType = undefined;
-        [ subType, params ] = this.handleSubTypeAndParams ('watchTickers', market, params);
+        [ subType, params ] = this.handleSubTypeAndParams ('watchTickers', firstMarket, params);
         if (this.isLinear (type, subType)) {
             type = 'future';
         } else if (this.isInverse (type, subType)) {
             type = 'delivery';
         }
-        const options = this.safeValue (this.options, 'watchTickers', {});
-        let name = this.safeString (options, 'name', 'ticker');
-        name = this.safeString (params, 'name', name);
-        params = this.omit (params, 'name');
-        let wsParams = [];
-        let messageHash = 'tickers';
-        if (symbols !== undefined) {
-            messageHash = 'tickers::' + symbols.join (',');
+        let name = undefined;
+        [ name, params ] = this.handleOptionAndParams (params, 'watchTickers', 'name', 'ticker');
+        const messageHashes = [];
+        if (this.isEmpty (symbols)) {
+            let markets = undefined;
+            if (type === 'spot') {
+                markets = this.filterBy (this.markets, 'type', type);
+            } else if (type === 'future') {
+                markets = this.filterBy (this.markets, 'subType', 'linear');
+            } else if (type === 'delivery') {
```

Comment:
why do we use old names? maybe we should everywhere use `this.linear()` and `this.inverse()`, otherwise, using old exchange specific names futures/delivery, etc make code unintuitive

Reply:
we could use `marketType` instead of `type` in above body:
```
[ marketType, params ] = this.handleMarketTypeAndParams ('watchTickers', firstMarket, params);
```
and then initialize separate variable `type`, so we can retain unified `marketType` value untouched (bcz inside if/else `type = ...` is being overwritten)